lib/src/message/line.rs

```diff
@@ -0,0 +1,79 @@
+// git-dit - the distributed issue tracker for git
+// Copyright (C) 2016, 2017 Matthias Beyer <mail@beyermatthias.de>
+// Copyright (C) 2016, 2017 Julian Ganz <neither@nut.email>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+
+use message::trailer::{Trailer, TrailerKey, TrailerValue};
+use regex::Regex;
+use std::iter::Peekable;
+use std::str::Lines as L;
+
+
+pub enum Line {
+    Text(String),
+    Trailer(Trailer),
+    Blank
+}
+
+impl<'a> From<&'a str> for Line {
+    fn from(line :&'a str) -> Self {
+        lazy_static! {
+            // TODO: do not accept whitespace in the key
+            static ref RE: Regex = Regex::new(r"^(?P<key>(.*)):\ (?P<value>(.*))$").unwrap();
+        }
+
+        let trimmed = line.trim_right();
+        if trimmed.is_empty() {
+            return Line::Blank;
+        }
+
+        if let Some(cap) = RE.captures(trimmed) {
+            if let (Some(key), Some(value)) = (cap.name("key"), cap.name("value")) {
+                return Line::Trailer(Trailer {
+                    key  : TrailerKey::from(String::from(key.as_str())),
+                    value: TrailerValue::from_slice(value.as_str()),
+                })
+            }
+        }
+
+        Line::Text(String::from(trimmed))
+    }
+}
+
+
+#[derive(Debug, Clone)]
```

Comment:
I am not sure whether we need `Clone` here. Also I'm not sure whether this would actually work as intended...